File: TravelPlannerApp/TravelService/Models/Activity.cs
using System;

namespace TravelService.Models
{
    public class Activity
    {
        public int Id { get; set; }
        public int TravelPlanId { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime Date { get; set; }
        public TimeSpan? Time { get; set; }
        public string Location { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal EstimatedCost { get; set; }
        public int Status { get; set; } // 0=Planned, 1=Reserved, 2=Completed, 3=Cancelled
        public DateTime CreatedAt { get; set; }
        public TravelPlan TravelPlan { get; set; }
    }
}

File: client/src/models/travel/Activity.ts
export interface Activity {
  id: number;
  travelPlanId: number;
  name: string;
  date: string;
  time: string | null;
  location: string;
  description: string;
  estimatedCost: number;
  status: ActivityStatus;
}

export enum ActivityStatus {
  Planned = 0,
  Reserved = 1,
  Completed = 2,
  Cancelled = 3,
}

export const ActivityStatusLabels: Record<ActivityStatus, string> = {
  [ActivityStatus.Planned]: "Planned",
  [ActivityStatus.Reserved]: "Reserved",
  [ActivityStatus.Completed]: "Completed",
  [ActivityStatus.Cancelled]: "Cancelled",
};

export const ActivityStatusColors: Record<ActivityStatus, string> = {
  [ActivityStatus.Planned]: "bg-blue-100 text-blue-700 border-blue-200",
  [ActivityStatus.Reserved]: "bg-amber-100 text-amber-700 border-amber-200",
  [ActivityStatus.Completed]: "bg-green-100 text-green-700 border-green-200",
  [ActivityStatus.Cancelled]: "bg-red-100 text-red-700 border-red-200",
};